test/integration/shoots/operations/clusterautoscaler.go

```diff
@@ -165,6 +166,125 @@ var _ = ginkgo.Describe("Shoot clusterautoscaler testing", func() {
 		framework.ExpectNoError(err)
 	}, testTimeout)
 
+	f.Beta().Serial().CIt("should autoscale a single worker group to/from zero", func(ctx context.Context) {
+		var (
+			shoot = f.Shoot
+
+			origClusterAutoscalerConfig = shoot.Spec.Kubernetes.ClusterAutoscaler.DeepCopy()
+			workerPoolName              = shoot.Spec.Provider.Workers[0].Name
+			testWorkerPoolName          = "ca-test"
+			origWorkers                 = shoot.Spec.Provider.Workers
+		)
+		if shoot.Spec.Provider.Type != "aws" && shoot.Spec.Provider.Type != "azure" {
+			ginkgo.Skip("not applicable")
+		}
+
+		// Create a dedicated worker-pool for cluster autoscaler.
+		testWorkerPool := shoot.Spec.Provider.Workers[0]
+		testWorkerPool.Name = testWorkerPoolName
+		testWorkerPool.Minimum = 0
+		testWorkerPool.Maximum = 2
+		testWorkerPool.Labels = map[string]string{"worker.gardener.cloud/pool": testWorkerPoolName}
+		testTaint := corev1.Taint{
+			Key:    testWorkerPoolName,
+			Effect: corev1.TaintEffectNoSchedule,
+			Value:  testWorkerPoolName,
+		}
+
+		testWorkerPool.Taints = append(testWorkerPool.Taints, testTaint)
+
+		ginkgo.By("updating shoot spec for test")
+		err := f.UpdateShoot(ctx, func(s *corev1beta1.Shoot) error {
+			s.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, testWorkerPool)
+
+			if s.Spec.Kubernetes.ClusterAutoscaler == nil {
+				s.Spec.Kubernetes.ClusterAutoscaler = &corev1beta1.ClusterAutoscaler{}
+			}
+			s.Spec.Kubernetes.ClusterAutoscaler.ScaleDownDelayAfterAdd = &metav1.Duration{Duration: scaleDownDelayAfterAdd}
+			s.Spec.Kubernetes.ClusterAutoscaler.ScaleDownUnneededTime = &metav1.Duration{Duration: scaleDownUnneededTime}
+
+			return nil
+		})
+		framework.ExpectNoError(err)
+
+		// Reset shoot spec to original values after test
+		defer func() {
```

Comment:
Sorry unrelated to this particular function, but a general issue.

@dguendisch @schrodit  can we make it possible in the framework to use the ginkgo.AfterSuite(). That is not possible any more because the framework already uses it to execute common functionality. 
I find it difficult to work around that and mostly end up cramping all the test code within on ginko.it() and separating it with ginko.By() (like in this case) - only to be able to execute cleanup functionality. 
I would either wish to have 
- an option to disable the framework to execute the ginko.AfterSuite()
- or the ability to register functions in a generic way that are also executed in the AfterSuite()

WDYT & is that feasible?

Reply:
Is there anything specific to be discussed here at current PR?

Reply:
I don't think so. Can we proceed with this PR @danielfoehrKn ?

Reply:
Go ahead please.

Reply:
I already thought of the second options and wanted to implement that as soon as it would be needed.
But in this specific use case the ginkgo aftersuite is not suitable as we want the cleanup after this test has finished and not the complete suite.
Now there are several tests running in parallel and even for serial tests the execution order is not controllable.
This means that aftersuite would not solve the problem as we want a clean environment after this test is completed, not after all tests.

Possible options then are:
- defer functions
- or `BeforeEach` in its own context

Does it make sense?

Reply:
yup, thats how I have done it so far. I see your point, just from my test framework "user perspective" it would be nice to be able to register normal Ginko.AfterSuites or a similar functionality provided by the test framework. 
Just FYI, maybe that could be improved at some point.